### PR TITLE
Create an initial PR template for DSpace 7 UI

### PR DESCRIPTION
**Pull request by tdonohue**
This PR creates a Pull Request template for all new PRs to `master` based on 
https://help.github.com/en/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository

The content here is based on the PR Template for the Backend (REST API), which was added in https://github.com/DSpace/DSpace/pull/2721

Feedback is welcome!  I created this in a shared branch so that others can help contribute to this PR easily.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+List of changes in this PR:
+* First, ...
+* Second, ...
+
+**Include guidance for how to test or review your PR.** This may include: steps to reproduce a bug, screenshots or description of a new feature, or reasons behind specific changes. 
+
+## Checklist
+_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
+
+- [ ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests). Exceptions may be made if previously agreed upon.
```
Comment (artlowel) [inline .github/pull_request_template.md]: I'd perhaps change this to

```suggestion
- [ ] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided an argument as to why that's not possible.
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- [ ] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
+- [ ] My PR passes all specs/tests and includes new/updated specs for any bug fixes, improvements or new features. A few reminders about what constitutes good tests:
+    * Include tests for different user types (if behavior differs), including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
+    * Include tests for error scenarios, e.g. when errors/warnings should appear (or buttons should be disabled).
+    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
+- [ ] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
```
Comment (artlowel) [inline .github/pull_request_template.md]: Perhaps we should start with a baseline check of all our licenses then, to ensure we're not breaking that rule already. That can be done using `npx license-checker --summary`

Comment (tdonohue) [reply .github/pull_request_template.md]: @artlowel : Thanks for noting that tool!  I've just run it and yes, all the existing licenses check out (there were a few I had to look up manually as they listed "Custom" as their license, but they checked out as well).  So, we're currently OK.

File: .github/pull_request_template.md (hunk 1 above)
Comment (tdonohue) [reply .github/pull_request_template.md]: I've updated the text similar to what you suggested. Thanks!

File: .github/pull_request_template.md (hunk 2 above)
Comment (artlowel) [reply .github/pull_request_template.md]: That's good news. Thanks for checking

**Pull request merged**